packages/client/src/components/UploadForm.tsx

```diff
@@ -285,19 +318,45 @@ export const UploadForm = () => {
 								<Label
 									className="text-base"
 									htmlFor="media-url"
-									value="Url(s) for transcription (one per line)"
+									value="URL(s) for transcription"
 								/>
 							</div>
 							<div className={'ml-3'}>
-								<Textarea
-									id="media-url"
-									placeholder="e.g. https://www.youtube.com?v=abc123"
-									required
-									rows={4}
-									onChange={(e) => {
-										setMediaUrlText(e.target.value);
-									}}
-								/>
+								{mediaUrlInputs.map((input, index) => (
+									<>
+										<TextInput
+											id={`media-url-${index}`}
+											placeholder="e.g. https://www.youtube.com?v=abc123"
+											className={'mt-1 mb-1'}
+											color={getStatusColor(input)}
+											helperText={
+												input.status === 'invalid' ? input.reason : ''
+											}
+											onChange={(e) => {
+												setMediaUrlInputs(
+													mediaUrlInputs.map((input, i) =>
+														i === index
+															? checkUrlInputValid({
+																	...input,
+																	value: e.target.value,
+																})
+															: input,
+													),
+												);
```

Comment:
Seems like this could be a bit simpler if we called a helper function that indexed into `mediaUrlInputs` rather than mapping
```typescript
	const setMediaUrlInput = (index: number, value: string) => {
		const newInputs = [...mediaUrlInputs];
		newInputs[index] = checkUrlInputValid(value);
		setMediaUrlInputs(newInputs);
	};
```
I think this is just a matter of personal preference though. Your way we avoid allocating a new array.

Reply:
I decided to do it inline but otherwise think this is a sensible improvement. I tend to try and keep things functional as if writing scala but it doesn't make sense given that I'm mutating stuff all over the place elsewhere!